sympy/geometry/line.py

```diff
@@ -1118,19 +1116,27 @@ def find(x_, equation_):
                     raise ValueError('ambiguous %s' % x_)
                 return xs[0]
 
+            xin, yin = x, y
+
             x = find(x, equation)
             y = find(y, equation)
 
-            if (len(solve(equation, y)) == 0) and (len(solve(equation, x)) != 0) and (degree(poly(equation, x)) == 1):
-                x_intercept = solve(equation, x)[0]
-                p1 = Point(x_intercept, 0)
-                p2 = Point(x_intercept, 1)
-                return Line(p1, p2)
-            else:
-                slope = diff(LC(poly(solve(equation, y)[0], y)), x)
-                p1 = (0, slope * 0)
-                p2 = (1, slope * 1)
-                return Line(p1, p2)
+            constant_term = equation.subs([(x, 0), (y, 0)])
```

Comment:
looks like you are anticipating input like `Eq(x/y, 3)` or equivalent. How about

```
...
equation = args[0]
if isinstance(equation, Equality):
    equation = equation.lhs - equation.rhs
n, d = [_mexpand(i, recursive=True) for i in equation.as_numer_denom()]
...
xin, yin = x, y
x = find(x, equation)
y = find(y, equation)
dfree = d.free_symbols
if x in dfree and y in dfree:
    # not linear
equation = n
a, b, c = linear_coeffs(equation, x, y)
```

You see that I am still suggesting `linear_coeffs`. What if someone calls `Line((x + 1)**2 + y)`?
```
constant_term = 1
x_coeff = 3
y_coeff = 0
```

Reply:
On second thought, such an expression really isn't strictly linear. In what context do you anticipate such input? (Also, my suggestion about x in dfree and y in dfree is wrong.)

Reply:
@smichr
I don't think there will be a need for this part :
```
if isinstance(equation, Equality):
    equation = equation.lhs - equation.rhs
n, d = [_mexpand(i, recursive=True) for i in equation.as_numer_denom()]
```

Although, the following part looks great : 
```
xin, yin = x, y
x = find(x, equation) or Dummy()
y = find(y, equation)  or Dummy()
a, b, c = linear_coeffs(eq, x, y)
if b:
    return Line((0,-c/b), slope=-a/b)
if a:
    return Line((-c/a, 0), slope=oo)
raise ValueError('neither %s nor %s were found in the equation' % (xin, yin))
```
It works fine now. (even on inputs like `Line((x + 1)**2 + y)` ). Thanks for that input.

Also, I've come to realise that inputs like `x/y` or `y/x` aren't exactly `Linear` 
(earlier, I assumed that an input of such a type :- 
` x / y = 0    ==>   x = 0 ` should be considered linear, where I was wrong)
Such an input now gives an error stating :- 
`ValueError: nonlinear term encountered: x/y`.